### PR TITLE
M-1 Merge to main

### PR DESCRIPTION
**Pull request by HunterJuneau**
### Description

We completed the READ part of CRUD, as well as authentication and the navbar.

### Features

- READ from CRUD
- Navbar
- Auth
- Basic Styling

### Related Issues

- #10 
- #11 
- #29 
- #20 
- #48 
- #24 
- #12 
- #9 

### Motivation and Context

Allows the user to read all of the data, as well as logging in and out and navigating the site.

### How can this be tested?

- https://nautilusexplorerdashboard.netlify.app/
OR
- `git fetch origin development`

### Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

**Comment by ajawashington** [comment]
Great Job Team! 

Those add buttons are still on the read-only dashboard, for unauthenticated users.

File: src/javascripts/components/pages/crew.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    document.querySelector('#crewPlaceholder').innerHTML += `
+    <div class="col-sm-3">
+      <div class="card v-25 m-3 p-4 shadow-lg rounded">
+        <h5 class="card-title">${member.firstname} ${member.lastname}</h5>
+        <img class="card-img-top px-2 shadow-lg rounded overflow-hidden" src=${member.image} alt="picture of ${member.firstname} ${member.lastname}">
+        <div class="card-body">
+          <span>${member.job}</span> <br>
+          <span>Tenure: ${member.months_tenure} months</span>
+          <hr>
+          ${user ? `<button class="btn btn-info shadow-lg" data-toggle="modal" data-target="#formModal" id="update-board-btn--${member.firebaseKey}">Update Info</button>
```
Comment (drteresavasquez) [inline src/javascripts/components/pages/crew.js:19]: Love the ternary here!! Y'all fancy!

File: src/javascripts/events/domEvents.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+import { showCrew, emptyCrew } from '../components/pages/crew';
+import getLogEntry from '../helpers/data/logEntryData';
+import { showLogEntry, emptyLogEntry } from '../components/pages/logEntry';
+import getSpecies from '../helpers/data/crudSpecies';
+import { showReadSpecies, noReadSpecies } from '../components/pages/species';
+import getDestinations from '../helpers/data/destinationsData';
+import destinationsView from '../components/pages/destinationsView';
+
+const domEvents = (user) => {
+  $('body').on('click', (e) => {
```
Comment (drteresavasquez) [inline src/javascripts/events/domEvents.js:12]: OK, Jquery!

File: src/javascripts/helpers/router.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+import 'firebase/auth';
+import loginButton from '../components/buttons/loginButton';
+import logoutButton from '../components/buttons/logoutButton';
+import firebaseConfig from './auth/apiKeys';
+import startApp from '../views/startApp';
+
+const checkLoginStatus = () => {
+  firebase.initializeApp(firebaseConfig);
+  firebase.auth().onAuthStateChanged((user) => {
+    startApp(user);
```
Comment (drteresavasquez) [inline src/javascripts/helpers/router.js:11]: Tell me about this being outside of the conditional. There is nothing wrong with this of course, I just would like to know how you came to do this 😄

**Review by drteresavasquez — COMMENTED** [review]
Just a few questions. @ajawashington can approve your PR.

File: src/javascripts/helpers/router.js (hunk 3 above)
Comment (HunterJuneau) [reply src/javascripts/helpers/router.js:11]: It allows the app to start regardless of whether the user is logged in or not. That way any user can READ only if they are unauthenticated.

Comment (drteresavasquez) [reply src/javascripts/helpers/router.js:11]: OK! I love it! So you are conditionally showing the buttons?

Comment (HunterJuneau) [reply src/javascripts/helpers/router.js:11]: Yes, just like in the ternary you pointed out above!

File: src/javascripts/events/navigationEvents.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    });
+  });
+
+  document.querySelector('#readSpecies').addEventListener('click', () => {
+    getSpecies(user).then((speciesArray) => {
+      if (speciesArray.length) {
+        showReadSpecies(speciesArray, user);
+      } else {
+        noReadSpecies();
+      }
```
Comment (ajawashington) [inline src/javascripts/events/navigationEvents.js:48]: So when clicking on the navbar as a authenticated user, I can see buttons for my `update` and `delete` however when on the dashboard and I click `Go There!` those buttons are not being rendered and this is for all resources.

**Review by ajawashington — CHANGES_REQUESTED** [review]
- [x] Remove buttons when an unauthenticated user is on read-only view.
- [x] Rendering different views and buttons for when an authenticated user click `Go There` on dashboard vs when clicking on Navbar

File: src/javascripts/events/navigationEvents.js (hunk 4 above)
Comment (HunterJuneau) [reply src/javascripts/events/navigationEvents.js:48]: Just pushed up a bug fix!

**Review by ajawashington — APPROVED** [review]
Great job team! You are approved!

**Pull request merged**